linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_files_permissions/bash/shared.sh

```diff
@@ -14,9 +14,19 @@ declare -a LOG_FILE_PATHS
 declare -a RSYSLOG_CONFIGS
 RSYSLOG_CONFIGS+=("${RSYSLOG_ETC_CONFIG}" "${RSYSLOG_INCLUDE_CONFIG[@]}" "${RSYSLOG_INCLUDE[@]}")
 
-# Browse each file selected above as containing paths of log files
-# ('/etc/rsyslog.conf' and '/etc/rsyslog.d/*.conf' in the default configuration)
-for LOG_FILE in "${RSYSLOG_CONFIGS[@]}"
+# Get full list of files to be checked
+# RSYSLOG_CONFIGS may contain globs such as 
+# /etc/rsyslog.d/*.conf /etc/rsyslog.d/*.frule
+# So, loop over the entries in RSYSLOG_CONFIGS and use find to get the list of included files.
+declare -a RSYSLOG_FILES
+for ENTRY in "${RSYSLOG_CONFIGS[@]}"
+do
+     mapfile -t FINDOUT < <(find "$(dirname "${ENTRY}")" -maxdepth 1 -name "$(basename "${ENTRY}")")
+     RSYSLOG_FILES=("${RSYSLOG_FILES[@]}" "${FINDOUT[@]}")
```

Comment:
```suggestion
     RSYSLOG_FILES+=("${FINDOUT[@]}")
```

I see that in the PR description there were mentions of duplicates when using `+=`, but that was probably caused by missing initialization of the array, so an earlier remediation influenced its contents. This proposed change should result in an exactly the same behavior, could you please comment on that?